src/app/services/auth.service.ts

```diff
@@ -18,22 +18,23 @@ export class AuthService {
   constructor(private http: HttpClient) { }
 
   register(registerRequest: RegisterRequest): Observable<AuthResponse>{
-      return this.http.post<AuthResponse>(`${this.BASE_URL}/register`,registerRequest)    
+      return this.http.post<AuthResponse>(`${this.BASE_URL}/register`,registerRequest).pipe(
+        catchError(this.handleError)
+      )   
   }
 
   login(loginRequest: LoginRequest): Observable<AuthResponse>{
-     return this.http.post<AuthResponse>(`${this.BASE_URL}/login`, loginRequest).pipe(
-      tap((response: AuthResponse) => {
-        if (response &&( response.accessToken || response.token)) {
-          if (typeof window !== 'undefined' && window.sessionStorage){
-
-            sessionStorage.setItem('token', response.accessToken || response.token)
-          }
-        } 
-      }),
-      catchError(this.handleError)
-    )
-  }
+    return this.http.post<AuthResponse>(`${this.BASE_URL}/login`, loginRequest).pipe(
```

Comment:
Consider storing your base URL in an environment variable.

Reply:
done